spec/index.bs

```diff
@@ -4215,48 +4215,58 @@ and {{GPUCommandEncoder/copyTextureToBuffer()}}.
-                GPUSize64 bytesPerImage = layout.bytesPerRow * layout.rowsPerImage;
-                GPUSize64 bytesInLastSlice =
-                    layout.bytesPerRow * (copyExtent.height / blockHeight - 1) + (copyExtent.width / blockWidth * blockSize);
-                requiredBytesInCopy = bytesPerImage * (copyExtent.depth - 1) + bytesInLastSlice;
-            }
-            ```
-
-    The following validation rules apply:
-
-    For the copy being in-bounds:
-    - If |layout|.{{GPUTextureDataLayout/rowsPerImage}} is not 0, it must be greater than or equal to
-        |copyExtent|.[=Extent3D/height=] &divide; |blockHeight|.
-    - (|layout|.{{GPUTextureDataLayout/offset}} + |requiredBytesInCopy|) must not overflow a {{GPUSize64}}.
-    - (|layout|.{{GPUTextureDataLayout/offset}} + |requiredBytesInCopy|) must be smaller than or equal to |byteSize|.
-
-    For the texel block alignments:
-    - |layout|.{{GPUTextureDataLayout/offset}} must be a multiple of |blockSize|.
-
-    For other members in |layout|:
-    - If |copyExtent|.[=Extent3D/height=] is greater than 1 or |copyExtent|.[=Extent3D/depth=] is greater than 1:
-        - |layout|.{{GPUTextureDataLayout/bytesPerRow}} must be greater than or equal to the number of |bytesInACompleteRow|.
-    - If |copyExtent|.[=Extent3D/depth=] is greater than 1:
-        - |layout|.{{GPUTextureDataLayout/rowsPerImage}} must not be 0.
+    : {{GPUTextureDataLayout}} |layout|
+    :: Layout of the linear texture data.
+    : {{GPUSize64}} |byteSize|
+    :: Total size of the linear data, in bytes.
+    : {{GPUTextureFormat}} |format|
+    :: Format of the texture.
+    : {{GPUExtent3D}} |copyExtent|
+    :: Extent of the texture to copy.
+
+    1. Let |blockWidth|, |blockHeight|, and |blockSize| be the
+        [=texel block width=], [=texel block height|height=], and
+        [=texel block size|size=] of |format|.
+
+    1. It is assumed that |copyExtent|.[=Extent3D/width=] is a multiple of |blockWidth|
+        and |copyExtent|.[=Extent3D/height=] is a multiple of |blockHeight|. Let:
+            - |widthInBlocks| be |copyExtent|.[=Extent3D/width=] &divide; |blockWidth|.
+            - |heightInBlocks| be |copyExtent|.[=Extent3D/height=] &divide; |blockHeight|.
+            - |bytesInACompleteRow| be |blockSize| &times; |widthInBlocks|.
+
+    1. Fail if the following conditions are not satisfied:
+        <div class=validusage>
+            - If |heightInBlocks| &gt; 1 or |copyExtent|.[=Extent3D/depth=] &gt; 1:
+                - |layout|.{{GPUTextureDataLayout/bytesPerRow}} must be
+                    greater than or equal to the number of |bytesInACompleteRow|.
+            - If |copyExtent|.[=Extent3D/depth=] &gt; 1:
+                - |layout|.{{GPUTextureDataLayout/rowsPerImage}} must not be 0.
+
+            These ensure the copy's subregions of the linear data do not overlap.
+        </div>
+
+    1. If |widthInBlocks|, |heightInBlocks|, or |copyExtent|.[=Extent3D/depth=] is 0:
```

Comment:
speaking of the Myles question on what you could do with zero-sized buffer: you can issue copy operations on it :)